src/modules/user/user.service.ts

```diff
@@ -109,13 +119,43 @@ export class UserService {
       return await this.userRepository.findOne({
         where: {
           email,
+          isDeleted: false,
         },
       });
     } catch (error) {
       throw new Error(`Error finding user: ${error.message}`);
     }
   }
 
+  async findAllByEmail(email: string): Promise<User[]> {
+    try {
+      return await this.userRepository.find({
+        where: {
+          email,
+        },
+      });
+    } catch (error) {
+      throw new Error(`Error finding users: ${error.message}`);
+    }
+  }
+
+  async findLatestActiveUserByEmail(email: string): Promise<User> {
+    try {
+      const users = await this.userRepository.find({
+        where: { email },
+        order: { createdAt: 'DESC' },
+      });
+
+      const activeUser = users.find((user) => !user.isDeleted);
+      if (!activeUser) {
+        throw new Error('Active user not found');
+      }
+      return activeUser;
+    } catch (error) {
+      throw new Error(`Error finding active user: ${error.message}`);
+    }
+  }
+
```

Comment:
Can't you just select active user from database without searching through array ?

Reply:
I did it this way to handle the situation when there is more than one user with same email and isDeleted: false. It's a hypothetical situation, and it's impossible if everything works right, but I would prefer to take it into account